spin/analytics.go

```diff
@@ -239,54 +218,28 @@ func (dc *dataCollection) doPrepData(btfsNode *core.IpfsNode) (*pb.SignedMetrics
 	return sm, errs, nil
 }
 
-func (dc *dataCollection) doSendData(ctx context.Context, config *config.Config, sm *pb.SignedMetrics) error {
+func (dc *dcWrap) doSendData(ctx context.Context, config *config.Config, sm *pb.SignedMetrics) error {
 	cb := cgrpc.StatusClient(config.Services.StatusServerDomain)
 	return cb.WithContext(ctx, func(ctx context.Context, client pb.StatusServiceClient) error {
 		_, err := client.UpdateMetrics(ctx, sm)
 		return err
 	})
 }
 
-func (dc *dataCollection) getPayload(btfsNode *core.IpfsNode) ([]byte, error) {
-	nd := new(node.Node)
-	now := time.Now().UTC()
-	nd.TimeCreated = now
-	nd.NodeId = dc.NodeID
-	nd.BtfsVersion = dc.BTFSVersion
-	nd.ArchType = dc.ArchType
-	nd.BlocksDown = dc.BlocksDown
-	nd.BlocksUp = dc.BlocksUp
-	nd.CpuInfo = dc.CPUInfo
-	nd.CpuUsed = dc.CPUUsed
-	nd.Download = dc.Download
-	nd.MemoryUsed = dc.MemUsed
-	nd.OsType = dc.OSType
-	nd.PeersConnected = dc.NumPeers
-	nd.StorageUsed = dc.StorageUsed
-	nd.UpTime = dc.UpTime
-	nd.Upload = dc.Upload
-	nd.TotalUpload = dc.TotalUp
-	nd.TotalDownload = dc.TotalDown
-	nd.HVal = dc.HVal
+func (dc *dcWrap) getPayload(btfsNode *core.IpfsNode) ([]byte, error) {
 	if config, err := dc.node.Repo.Config(); err == nil {
```

Comment:
config does not change without restarting daemon, so you can safely set this inside `Analytics()`

Reply:
okay